src/ai/defaultai.cc

```diff
@@ -2926,6 +2971,18 @@ bool DefaultAI::construct_building(const Time& gametime) {
 					number_of_same_nearby += same_it->second;
 				}
 
+				// Considering distance to wh for various types of productionsites:
+				if (bo.is(BuildingAttribute::kWell) || bo.is(BuildingAttribute::kRanger) ||
+				    bo.is(BuildingAttribute::kHunter)) {
+					prio += wh_distance_malus;  // push farer
+				} else if (bo.is(BuildingAttribute::kRecruitment)) {
+					prio -= wh_distance_malus;  // push closer
+				} else if (bo.is(BuildingAttribute::kSpaceConsumer)) {
+					prio += wh_distance_malus;  // push farer
+				} else if (!bo.inputs.empty() && !bo.ware_outputs.empty() && !bo.requires_supporters) {
```

Comment:
here we should include the recruitment site for codecheck

Reply:
Yes, exactly what I meant (both changes)